pkg/segment/context/context.go

```diff
@@ -22,16 +33,49 @@ func NewContext(ctx context.Context) context.Context {
 
 // GetContextProperties retrieves all the values set in a given context
 func GetContextProperties(ctx context.Context) map[string]interface{} {
-	properties := propertiesFromContext(ctx)
-	if properties == nil {
+	cProperties := propertiesFromContext(ctx)
+	if cProperties == nil {
 		return make(map[string]interface{})
 	}
-	return properties.values()
+	return cProperties.values()
 }
 
-// SetComponentType sets componentType property for telemetry data when a new component is created
+// SetComponentType sets componentType property for telemetry data when a component is created/pushed
 func SetComponentType(ctx context.Context, value string) {
-	setContextProperty(ctx, "componentType", value)
+	setContextProperty(ctx, ComponentType, util.ExtractComponentType(value))
+}
+
+// SetClusterType sets clusterType property for telemetry data when a component is pushed or a project is created/set
+func SetClusterType(ctx context.Context, client *occlient.Client) {
+	var value string
+	if client == nil {
+		value = "not-found"
+	} else {
+		// We are not checking ServerVersion to decide the cluster type because it does not always return the version,
+		// it sometimes fails to retrieve the data if user is using minishift or plain oc cluster
```

Comment:
`or plain oc cluster`

What's a plain oc cluster?

Reply:
It means the basic cluster that is created on running `oc cluster up`.